More numpythonic osc funcs, now w/ more args (phase, amp, duty cycle)

jbdoar opened the pull request:
Fixes # .

Changes proposed in this pull request: 
- 
-
-

Mentions: @alxrsngrtn @jbdoar

jbdoar committed:
More numpythonic osc funcs, now w/ more args (phase, amp, duty cycle)

jbdoar committed:
adding functions for low and highpass filters (4th order with resonance)

alxmrs reviewed (approved):
Looks good. 

Nit: Instead of using single-line comments for low and hi pass, use a """docstring""".

jbdoar committed:
deleting extraneous files

jbdoar committed:
deleting extraneous files

jbdoar committed:
deleting extraneous files

jbdoar committed:
deleting extraneous files

jbdoar committed:
mashed all synthesizer related files up in here together

jbdoar committed:
deleting VCO function (new defs for the oscillators work OK without it)

jbdoar committed:
removing crud, adding different crud (see sample & hold, envelope)

jbdoar committed:
cleaning things up...

jbdoar committed:
Adding quantize_frequency back into library

jbdoar committed:
fixed indent error in quantize_frequency

jbdoar committed:
hardcoding default sample rate into filter functions, at least for now

jbdoar committed:
editing init file

jbdoar committed:
edit to time function

jbdoar committed:
updated import stuff

jbdoar committed:
didn't work if T wasn't an int

alxmrs reviewed (changes requested):
Before merge, please make sure `bin/bish` still works.

jbdoar commented:
> Before merge, please make sure `bin/bish` still works.

Good idea. Do you have any minimal working examples of how to get that part of the project to work? I haven't figured it out yet.

alxrsngrtn commented:
Yes -- I believe examples should be documented in the help message of the script. So, running `bin/bish --help` will do the trick. I think you will hit import errors before then.

jbdoar commented:
Well, flattening the file structure of the hcm library certainly has introduced some errors--we certainly won't wish to merge anything yet.

jbdoar commented:
Alright, so: when I try running bish, I get this error, flagging line 39 in the io module:
`AttributeError:` module 'rx' has no attribute 'Observer'

Any thoughts on where that might be coming from? In case it matters, I have rx version 3.1.1 installed on my machine, which differs from the version in your requirements.txt.

alxrsngrtn commented:
Ah. So the rx api has changed from our reqs to the latest release. The easy fix would be to uninstall rx and `pip install -r requirements.txt`. We also could refactor our use of rx in the project to use the latest api. The latter is a lot of effort, so I'm ok with doing the former.

jbdoar committed:
updated function calls to reflect flattened synth module structure

jbdoar committed:
linspace in python 3 requires num be an int

I think we can get rid of VCO now, but it's still in bish and I didn't 
want to delete it for that reason

jbdoar committed:
couldn't find normalize otherwise

jbdoar commented:
Will need more reviewing before merge. Presently I am no longer getting errors when running bish, but it's not driving my speakers, either...

alxrsngrtn commented:
hmm... it looks like `bin/bish` was deleted in this branch. How should I test it?

jbdoar committed:
Add files via upload

dunno how it disappeared from the branch

alxrsngrtn commented on hcm/bish:
```diff
@@ -11,7 +11,7 @@ from hcm import types
 SAMPLE_RATE = 8000  # hz
 INTERVAL_LENGTH = 1000  # ms
 
```
I see you moved the file from `bin`. Please update `setup.py` to point to this new location of bish: 
```
scripts=['hcm/bish']
```

jbdoar replied:
I'm sure there was a better way than moving bish into the main directory, but for whatever reason it wasn't finding hcm when I tried running it out of the bin folder.

alxrsngrtn commented:
> it's not driving my speakers

I see. I filed a bug for this, I can look into and fix it. Once you update `setup.py`, this PR is good to go.

alxmrs reviewed (approved):
LGTM, after minor change to setup.py.

alxmrs committed:
Fixed broken `bish` script in install. (#25)

alxmrs committed:
just import `hcm` and project management with rye/pyproject.toml. (#26)

alxmrs committed:
different minimum version of python.

alxmrs committed:
Cleanup & Updated docs. (#27)

alxmrs committed:
Created using Colab

alxmrs committed:
Halfway through the notebook

alxmrs commented:
Closing pull request. I just did a diff between with this and `main` (originally, this PR is into `master`), and there is no difference! I'm happy to see that the desired effects have been made.
